### ISSUE TITLE
Legacy Solve not Working

### ISSUE THREAD

**Issue by Vivekanandhan19** [david-cattermole/mayaMatchMoveSolver#229]
Legacy Solve not Working
During Solve Legacy Solve not Working. -->

**Expected behaviour:**

**Actual behaviour:**

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...

Maya 2018.6

- mmSolver version: [v0.3.14]

- Maya version:
Maya 2018.6
Use the menu "Help > About Maya", or run the following Python commands in the Maya Script Editor.

print 'Operating System:', maya.cmds.about(operatingSystemVersion=True)
print 'Cut ID:', maya.cmds.about(cutIdentifier=True)
print 'Maya Version:', maya.cmds.about(installedVersion=True)
print 'Maya API:', maya.cmds.about(apiVersion=True)
print 'Qt Version:', maya.cmds.about(qtVersion=True)
print 'Compositing Manager:', maya.cmds.about(compositingManager=True)
print 'Window Manager:', maya.cmds.about(windowManager=True)
print 'GPU:', ''.join(maya.cmds.ogs(deviceInformation=True))
-->

[Maya version number or print out here]

- Operating System (OS): [Possible options: Windows, Linux and MacOS]

**Comment by david-cattermole** [comment]
Hello Vivekanandhan19,

Can you please give more details about your problem?

What settings are you using?

Do you get any errors in the Script Editor?

How is the Legacy solver not working?

Can you provide steps to re-create the problem? Or a scene file?

Also, the Legacy solver is named "legacy" because it is very old, and has
been superseded by the Basic and Standard solvers. Have you tried using
those instead?

David

**Comment by david-cattermole** [comment]
Hello @Vivekanandhan19,

I assume you're perhaps getting an error such as:
```
# Traceback (most recent call last):
#   File "C:\Users\username\Documents\maya\2019\modules\mayaMatchMoveSolver-0.3.14-maya2019-win64\python\mmSolver\tools\solver\ui\solver_window.py", line 696, in apply
#     self)
#   File "C:\Users\username\Documents\maya\2019\modules\mayaMatchMoveSolver-0.3.14-maya2019-win64\python\mmSolver\tools\solver\lib\collection.py", line 810, in run_solve_ui
#     info_fn=info_fn,
#   File "C:\Users\username\Documents\maya\2019\modules\mayaMatchMoveSolver-0.3.14-maya2019-win64\python\mmSolver\tools\solver\lib\collection.py", line 653, in execute_collection
#     info_fn=info_fn,
#   File "C:\Users\username\Documents\maya\2019\modules\mayaMatchMoveSolver-0.3.14-maya2019-win64\python\mmSolver\_api\execute.py", line 736, in execute
#     vaction_state_list = _run_validate_action_list(vaction_list)
#   File "C:\Users\username\Documents\maya\2019\modules\mayaMatchMoveSolver-0.3.14-maya2019-win64\python\mmSolver\_api\execute.py", line 537, in _run_validate_action_list
#     state = _run_validate_action(vaction)
#   File "C:\Users\username\Documents\maya\2019\modules\mayaMatchMoveSolver-0.3.14-maya2019-win64\python\mmSolver\_api\execute.py", line 476, in _run_validate_action
#     solve_data = vfunc(*vargs, **vkwargs)
#   File "<string>", line 2, in mmSolver
# TypeError: Invalid arguments for flag 'removeUnusedMarkers'.  Expected int, got NoneType
```

If this is easy to fix, I will fix it, otherwise I will remove the solver.
The Legacy solver was planned to be removed in an upcoming release anyway.

I recommend using the _Basic_ and _Standard_ tabs, as they replace the same features as the Legacy solver.
https://david-cattermole.github.io/mayaMatchMoveSolver/tools_solver_ui_tab_basic.html
https://david-cattermole.github.io/mayaMatchMoveSolver/tools_solver_ui_tab_standard.html

**Comment by Vivekanandhan19** [comment]
Hi David,

Thanks for the reply. Also in the standard root frames solver, which will
give better results either auto selected root frames or given root frames
based on character main key poses. It will be really good if we have a demo
video on the root frames function.

**Comment by david-cattermole** [comment]
Hello @Vivekanandhan19,

The link above is probably the best description of the features of the Standard tab.

A practical example of using mmSolver for a non-trivial solve, including refinement and using the Standard solver can be found here:
https://youtu.be/y9U2QZT-VxA

If you De not solving Static attribute values, it doesn't make much sense to use the Standard solver, and tve Basic solver will likely work best for you.

With the Standard solver I usually manually set the Root frames based on how I know the solver works. Sometimes I use auto as a starting point then manually edit the Root frame list.

To fully understand the way the solver works, see this:
https://david-cattermole.github.io/mayaMatchMoveSolver/solver_design.html

If you have other questions, please use the Mailing list, or open a new GitHub issue.
https://groups.google.com/g/maya-matchmove-solver

Thanks,
David

**Comment by david-cattermole** [comment]
Hello @Vivekanandhan19,

Thanks for reporting :)

This issue has been fixed and will be released in v0.3.15.

I would still highly recommend using the Basic or Standard solver tabs, and not using the Legacy tab, because it will be removed in a future version and is not well tested.

David
